lib/src/simple_config_origin.cc

```diff
@@ -1,34 +1,28 @@
 #include <internal/simple_config_origin.hpp>
+#include <stdexcept>
 
 using namespace std;
 
 namespace hocon {
 
     simple_config_origin::simple_config_origin(string description, int line_number, int end_line_number,
-                                               origin_type org_type, string resource_or_null,
-                                               vector<string> comments_or_null) {
-        if (description.empty()) {
+                                               origin_type org_type, string resource, vector<string> comments) :
+        _description(move(description)), _line_number(line_number), _end_line_number(end_line_number),
+        _origin_type(org_type), _resource_or_null(move(resource)), _comments_or_null(move(comments))
```

Comment:
We'll have to test out cases like that. boost::optional is one way to deal with that (or using unique pointers).